### PR TITLE
Present master services agreement

### PR DESCRIPTION
**Pull request by davidmreed**
MetaDeploy now supports presenting a Master Services Agreement (MSA) prior to product-specific terms, when a user requests to create a scratch org.

### CODE REVIEW

**Comment by davidmreed** [comment]
Ready for code review, but pending discussion with Legal to confirm this implementation meets requirements.

**Review by prescod — COMMENTED** [review]
I couldn't find any issues with the code itself, but I also do not have a ton of context for the UI flow, so take it with a grain of salt. Let me know if you need an Approval to kick off some other step in the process.

**Review by pconrad-sfdo — APPROVED** [review]

File: locales_dev/en/translation.json [hunk 1]
```diff
@@ -59,6 +60,7 @@
   "Log Out to Create Scratch Org": "Log Out to Create Scratch Org",
   "Log Out to Use Scratch Org": "Log Out to Use Scratch Org",
   "Log in": "Log in",
+  "Master Services Agreement": "Master Services Agreement",
```
Comment (davisagli) [inline locales_dev/en/translation.json:63]: This needs to be added to locales/en/translation.json as well, and we need to remember to submit it for translation prior to label freeze.

File: src/js/components/scratchOrgs/spinOrgModal.tsx [hunk 2]
```diff
@@ -18,13 +18,29 @@ const SpinOrg = ({
   toggleModal,
   doSpinOrg,
 }: Props) => {
+  const masterAgreement = window.GLOBALS.SITE?.master_agreement;
   const [confirmed, setConfirmed] = useState(!clickThroughAgreement);
-  const [currentPage, setCurrentPage] = useState(clickThroughAgreement ? 0 : 1);
+  const [msaConfirmed, setMsaConfirmed] = useState(!masterAgreement);
+  let startPage = 2;
```
Comment (davisagli) [inline src/js/components/scratchOrgs/spinOrgModal.tsx:24]: Could you use constants or members of a JS object to refer to the integers, to make this more readable?

File: metadeploy/api/models.py [hunk 3]
```diff
@@ -1110,6 +1110,7 @@ class SiteProfile(TranslatableModel):
         name=models.CharField(max_length=64),
         company_name=models.CharField(max_length=64, blank=True),
         welcome_text=MarkdownField(),
+        master_agreement=MarkdownField(),
```
Comment (davisagli) [inline metadeploy/api/models.py:1113]: Maybe include help text indicating that this is only shown for scratch orgs.

Do we need to add and maintain copies of the MSA in multiple languages?

File: src/js/components/scratchOrgs/spinOrgModal.tsx [hunk 4]
```diff
@@ -18,13 +18,29 @@ const SpinOrg = ({
...
+  }
+  const [currentPage, setCurrentPage] = useState(startPage);
   const [email, setEmail] = useState('');
 
   const nextPage = useCallback(() => {
-    setCurrentPage(currentPage + 1);
-  }, [currentPage]);
+    switch (currentPage) {
+      case 0:
+        setCurrentPage(confirmed ? 2 : 1);
```
Comment (Br4nd0R) [inline src/js/components/scratchOrgs/spinOrgModal.tsx:37]: Could `confirmed` be `clickThroughConfirmed` instead? Or comments?

**Review by Br4nd0R — APPROVED** [review]

File: metadeploy/api/models.py (hunk 3 above)
Comment (davidmreed) [reply metadeploy/api/models.py:1133]: Added. I do not know if Legal is going to ask for translated MSAs or not.

File: src/js/components/scratchOrgs/spinOrgModal.tsx (hunk 2 above)
Comment (davidmreed) [reply src/js/components/scratchOrgs/spinOrgModal.tsx:24]: 👍 Good callout, this logic is pretty confusing.

File: src/js/components/scratchOrgs/spinOrgModal.tsx (hunk 4 above)
Comment (davidmreed) [reply src/js/components/scratchOrgs/spinOrgModal.tsx:37]: Added comments, I'm weak enough on React that I'm slightly apprehensive of making unneeded changes.

**Review by davisagli — APPROVED** [review]

**Pull request merged**